plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/kotlin/KSTypeImpl.kt

```diff
@@ -10,24 +10,30 @@ import org.jetbrains.kotlin.ksp.symbol.*
 import org.jetbrains.kotlin.ksp.symbol.impl.KSObjectCache
 import org.jetbrains.kotlin.ksp.symbol.impl.binary.KSTypeArgumentDescriptorImpl
 import org.jetbrains.kotlin.ksp.symbol.impl.replaceTypeArguments
+import org.jetbrains.kotlin.types.AbbreviatedType
 import org.jetbrains.kotlin.types.KotlinType
 import org.jetbrains.kotlin.types.typeUtil.*
 
 class KSTypeImpl private constructor(
     val kotlinType: KotlinType,
+    val abbreviation: KotlinType?,
     private val ksTypeArguments: List<KSTypeArgument>? = null,
     override val annotations: List<KSAnnotation> = listOf()
 ) : KSType {
-    companion object : KSObjectCache<KotlinType, KSTypeImpl>() {
+    companion object : KSObjectCache<Pair<KotlinType, KotlinType?>, KSTypeImpl>() {
         fun getCached(
             kotlinType: KotlinType,
             ksTypeArguments: List<KSTypeArgument>? = null,
             annotations: List<KSAnnotation> = listOf()
-        ) =
-            cache.getOrPut(kotlinType) { KSTypeImpl(kotlinType, ksTypeArguments, annotations) }
+        ): KSTypeImpl {
+            val abbrev = (kotlinType as? AbbreviatedType)?.abbreviation
```

Comment:
There is an extension function `fun KotlinType.getAbbreviation(): SimpleType? = getAbbreviatedType()?.abbreviation` which looks a little bit different than this (calling unwrap() before casting), have you checked that?

Reply:
Yes unwrap() is safer. Thanks for catching that.